tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/CircuitBreakerTest.java

```diff
@@ -44,244 +46,469 @@
+    }
+
+    /**
+     * Analagous to testCircuitClosedThenOpen but with a Class level annotation
+     * specified that is overriden by a Method level annotation on serviceC.
+     * 
+     * With successThreshold = 2, requestVolumeThreshold = 2, failureRatio=1,
+     * delay = 50000 the expected behaviour is,
+     * 
+     * Execution Behaviour 
+     * ========= ========= 
+     * 1         RunTimeException 
+     * 2         RunTimeException 
+     * 3         CircuitBreakerOpenException  
+     * 4         CircuitBreakerOpenException 
+     * 5         CircuitBreakerOpenException 
+     * 6         CircuitBreakerOpenException 
+     * 7         CircuitBreakerOpenException
+     */
+    @Test
+    public void testClassLevelCircuitOverride() {
+        for (int i = 0; i < 7; i++) {
+            try {
+                clientForClassLevelCBWithDelay.serviceC();
+
+                if (i < 2) {
+                    Assert.fail("serviceC should throw an Exception in testClassLevelCircuitOverride");
+                }
+            } 
+            catch (CircuitBreakerOpenException cboe) {
+                // Expected on iteration 4
+                if (i < 2) {
+                    Assert.fail("serviceC should throw a RuntimeException in testClassLevelCircuitOverride");
+                }
+            } 
+            catch (RuntimeException ex) {
+                // Expected
+            } 
+            catch (Exception ex) {
+                // Not Expected
+                Assert.fail("serviceC should throw a RuntimeException in testClassLevelCircuitOverride");
+            }
+        }
+
+        int serviceAExecutions = clientForClassLevelCBWithDelay.getCounterForInvokingService();
+
+        Assert.assertEquals(serviceAExecutions, 2, "The number of executions should be 2");
+    }
+
+    /**
+     * Analagous to testCircuitReClose but with a Class level annotation
+     * specified that is overriden by a Method level annotation on serviceD.
+     * 
+     * With successThreshold = 2, requestVolumeThreshold = 4, failureRatio=0.75,
+     * delay = 1 the expected behaviour is,
+     * 
+     * Execution Behaviour 
+     * ========= ========= 
+     * 1         RunTimeException 
+     * 2         RunTimeException 
+     * 3         RunTimeException  
+     * 4         RunTimeException
+     * Pause for longer than CircuitBreaker delay, so that it transitions to half-open
+     * 5         SUCCEED 
+     * 6         SUCCEED (CircuitBreaker will be re-closed as successThreshold is 2)
+     * 7         SUCCEED
+     */
+    @Test
+    public void testClassLevelCircuitOverrideNoDelay() {
+        for (int i = 0; i < 7; i++) {
+            try {
+                // Pause to allow the circuit breaker to half-open on iteration
+                // 4
+                // This is conservative when the circuit breaker has a minimal
+                // delay.
+                if (i == 4) {
+                    try {
+                        Thread.sleep(500);
```

Comment:
+1 on the idea. However, I would like to pause the introduction of another open source project until MP 1.1 has been finally released. Otherwise, the addition of another open source project might delay the release. 

I raised [issue 37](https://github.com/eclipse/microprofile-fault-tolerance/issues/73) to update the current code base as this file is not the only place we used sleep().